src/librustc_privacy/lib.rs

```diff
@@ -826,7 +948,12 @@ impl DefIdVisitor<'tcx> for ReachEverythingInTheInterfaceVisitor<'_, 'tcx> {
     fn tcx(&self) -> TyCtxt<'tcx> { self.ev.tcx }
     fn visit_def_id(&mut self, def_id: DefId, _kind: &str, _descr: &dyn fmt::Display) -> bool {
         if let Some(hir_id) = self.ev.tcx.hir().as_local_hir_id(def_id) {
-            self.ev.update(hir_id, self.access_level);
+            if let ((ty::Visibility::Public, ..), _)
+                | (_, Some(AccessLevel::ReachableFromImplTrait))
```

Comment:
So, the consequence is that we don't mark `Priv` as reachable in situations like this:
```rust
pub fn foo()  where Priv: Condition { ... }
```
This case is not covered by type privacy and `foo` can be used from other crates.
Is `Priv` required to be reachable at link time in this case? I think no, so what this PR does is ok.
Anyway, if something breaks we'll get bug reports and add them into the test suite.